src/components/Pressable/Pressable.tsx

```diff
@@ -0,0 +1,247 @@
+                event.changedTouches.at(-1)
+              ) ||
+              isPressedDown.current
+            ) {
+              return;
+            }
+
+            isPressedDown.current = true;
+
+            if (props.unstable_pressDelay) {
+              pressDelayTimeoutRef.current = setTimeout(() => {
+                pressInHandler(event);
+              }, props.unstable_pressDelay);
+            } else {
+              pressInHandler(event);
+            }
+          });
+        })
+        .onTouchesUp((event) => {
+          if (
+            !isPressedDown.current ||
+            event.allTouches.length > event.changedTouches.length
+          ) {
+            return;
+          }
+
+          if (
+            props.unstable_pressDelay &&
+            pressDelayTimeoutRef.current !== null
+          ) {
+            // if pressDelay is set, we want to call onPressIn on touch up
+            clearTimeout(pressDelayTimeoutRef.current);
+            pressInHandler(event);
+          }
+
+          props.onPressOut?.(adaptTouchEvent(event));
+
+          if (isPressCallbackEnabled.current) {
+            props.onPress?.(adaptTouchEvent(event));
+          }
+
+          isPressedDown.current = false;
+          setPressedState(false);
+        })
+        .onTouchesCancelled((event) => {
+          if (
+            !isPressedDown.current ||
+            event.allTouches.length > event.changedTouches.length
+          ) {
+            return;
+          }
+
+          // original triggers onPressOut on cancel, but not onPress
+          props.onPressOut?.(adaptTouchEvent(event));
+
+          isPressedDown.current = false;
+          setPressedState(false);
+        }),
+    [
+      props.onPress,
+      props.onPressIn,
+      props.onPressOut,
+      setPressedState,
+      isPressedDown,
+      isPressCallbackEnabled,
+      normalizedHitSlop,
+      pressDelayTimeoutRef,
+    ]
+  );
+
+  // rippleGesture lives inside RNButton to enable android's ripple
+  const rippleGesture = useMemo(() => Gesture.Native(), []);
+
+  pressGesture.minDuration(
+    (props.delayLongPress ?? DEFAULT_LONG_PRESS_DURATION) +
+      (props.unstable_pressDelay ?? 0)
+  );
+
+  const appliedHitSlop = addInsets(
+    normalizedHitSlop,
+    normalizedPressRetentionOffset
+  );
+
+  const isPressableEnabled = props.disabled !== true;
+
+  touchGesture.enabled(isPressableEnabled);
+  pressGesture.enabled(isPressableEnabled);
+  hoverGesture.enabled(isPressableEnabled);
+
+  touchGesture.shouldCancelWhenOutside(true);
+  pressGesture.shouldCancelWhenOutside(true);
+  hoverGesture.shouldCancelWhenOutside(true);
+
+  touchGesture.runOnJS(true);
+  pressGesture.runOnJS(true);
+  hoverGesture.runOnJS(true);
+  rippleGesture.runOnJS(true);
+
+  // this hitSlop block is required by android
+  touchGesture.hitSlop(appliedHitSlop);
+  pressGesture.hitSlop(appliedHitSlop);
+  hoverGesture.hitSlop(appliedHitSlop);
+  rippleGesture.hitSlop(normalizedHitSlop);
```

Comment:
Why does `rippleGesture` use different `hitSlop`?

Reply:
So that ripple doesn't activate when you tap on the press retention area

Reply:
I'll do that.